src/litserve/server.py

```diff
@@ -472,3 +472,94 @@ def setup_auth(self):
+    async with AsyncExitStack() as stack:
+        for server in servers:
+            await stack.enter_async_context(server.lifespan(server.app))
+        yield
+
+
+def run_all(
+    servers: List[LitServer],
+    port: Union[str, int] = 8000,
+    num_api_servers: Optional[int] = 1,
+    log_level: str = "info",
+    generate_client_file: bool = True,
+    api_server_worker_type: Optional[str] = None,
+    **kwargs,
+):
+    """Run multiple LitServers on the same port."""
+
+    if any(not isinstance(server, LitServer) for server in servers):
+        raise ValueError("All elements in the servers list must be instances of LitServer")
+
+    if generate_client_file:
+        LitServer.generate_client_file()
+
+    port_msg = f"port must be a value from 1024 to 65535 but got {port}"
+    try:
+        port = int(port)
+    except ValueError:
+        raise ValueError(port_msg)
+    if not (1024 <= port <= 65535):
+        raise ValueError(port_msg)
+
+    if num_api_servers < 1:
+        raise ValueError("num_api_servers must be greater than 0")
+
+    if sys.platform == "win32":
+        print("Windows does not support forking. Using threads api_server_worker_type will be set to 'thread'")
+        api_server_worker_type = "thread"
+    elif api_server_worker_type is None:
+        api_server_worker_type = "process"
+
+    # Create the main FastAPI app
+    app = FastAPI(lifespan=lambda app: multi_server_lifespan(app, servers))
+    config = uvicorn.Config(app=app, host="0.0.0.0", port=port, log_level=log_level, **kwargs)
+    sockets = [config.bind_socket()]
+
+    managers, inference_workers = [], []
+    try:
+        for server in servers:
+            manager, workers = server.launch_inference_worker(num_api_servers)
+            managers.append(manager)
+            inference_workers.extend(workers)
+
+            # include routes from each litserver's app into the main app
+            app.include_router(server.app.router)
+
+        server_processes = []
+        for response_queue_id in range(num_api_servers):
+            for server in servers:
+                server.app.response_queue_id = response_queue_id
+                if server.lit_spec:
+                    server.lit_spec.response_queue_id = response_queue_id
+
+            app = copy.copy(app)
+            config = uvicorn.Config(app=app, host="0.0.0.0", port=port, log_level=log_level, **kwargs)
+            uvicorn_server = uvicorn.Server(config=config)
+
+            if api_server_worker_type == "process":
+                ctx = mp.get_context("fork")
+                worker = ctx.Process(target=uvicorn_server.run, args=(sockets,))
+            elif api_server_worker_type == "thread":
+                worker = threading.Thread(target=uvicorn_server.run, args=(sockets,))
+            else:
+                raise ValueError("Invalid value for api_server_worker_type. Must be 'process' or 'thread'")
+            worker.start()
+            server_processes.append(worker)
```

Comment:
To address the repetition in this section, I will create an issue to be handled in the next PR. I believe we could utilize the existing _start_server function from LitServer with some modifications—perhaps a static version.
cc: @aniketmaurya